dashboard/app/models/user.rb

```diff
@@ -2179,8 +2194,12 @@ def user_storage_id
   def undestroy
     raise 'Unable to restore a purged user' if purged_at
 
+    soft_delete_time = deleted_at
+
     # Paranoia documentation at https://github.com/rubysherpas/paranoia#usage.
-    restore(recursive: true, recovery_window: 5.minutes)
+    result = restore(recursive: true, recovery_window: 5.minutes)
+    restore_channels_deleted_after(soft_delete_time - 5.minutes)
```

Comment:
Projects are soft-deleted _before_ the User is soft-deleted, so I'm restoring them after the User is restored and keeping the 5-minute window.